charts/metrics-server/templates/apiservice.yaml

```diff
@@ -1,17 +1,46 @@
-{{- if .Values.apiService.create -}}
+{{- $altNames := list }}
+{{- $certs := dict }}
+
+{{- if eq .Values.tls.type "helm" }}
+{{- $commonName := include "metrics-server.fullname" . }}
+{{- $ns := .Release.Namespace }}
+{{- $altNames = append $altNames (printf "%s.%s" $commonName $ns) }}
+{{- $altNames = append $altNames (printf "%s.%s.svc" $commonName $ns) }}
+{{- $altNames = append $altNames (printf "%s.%s.svc.%s" $commonName $ns .Values.tls.clusterDomain) }}
+{{- $certs = genSelfSignedCert $commonName nil $altNames 36500 }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "metrics-server.fullname" . }}
+  labels:
+    {{- include "metrics-server.labels" . | nindent 4 }}
+type: Opaque
+data:
+  tls.crt: {{ $certs.Cert| b64enc | quote }}
+  tls.key: {{ $certs.Key | b64enc | quote }}
+{{- end }}
+---
```

Comment:
Please could you move this to _certificate.yaml_ and gate it behind the Helm type.

Reply:
Ahhh yes, I'd missed the use of the CA lower down.